docs/concepts/projects/dependencies.md

```diff
@@ -169,128 +185,150 @@ dependencies = [
 ]
 ```
 
-If the project only requires packages from standard package indexes, then `project.dependencies` is
-sufficient. If the project depends on packages from Git, remote URLs, or local sources,
-`tool.uv.sources` can be used to enrich the dependency metadata without ejecting from the
-standards-compliant `project.dependencies` table.
-
 ## Dependency sources
 
-During development, a project may rely on a package that isn't available on PyPI. The following
-additional sources are supported by uv:
+The `tool.uv.sources` table extends the standard dependency tables with alternative dependency
+sources, which are used during development.
 
-- Index: A package resolved from a specific package index.
-- Git: A Git repository.
-- URL: A remote wheel or source distribution.
-- Path: A local wheel, source distribution, or project directory.
-- Workspace: A member of the current workspace.
+Dependency sources add support common patterns that are not supported by the `project.dependencies`
+standard, like editable installations and relative paths. For example, to install `foo` from a
+directory relative to the project root:
 
-Note that if a non-uv project uses a project with sources as a Git- or path-dependency, only
-`project.dependencies` and `project.optional-dependencies` are respected. Any information provided
-in the source table will need to be re-specified in a format specific to the other package manager.
-
-To instruct uv to ignore the `tool.uv.sources` table (e.g., to simulate resolving with the package's
-published metadata), use the `--no-sources` flag:
+```toml title="pyproject.toml" hl_lines="7"
+[project]
+name = "example"
+version = "0.1.0"
+dependencies = ["foo"]
 
-```console
-$ uv lock --no-sources
+[tool.uv.sources]
+foo = { path = "./packages/foo" }
 ```
 
-The use of `--no-sources` will also prevent uv from discovering any
-[workspace members](#workspace-member) that could satisfy a given dependency.
+The following dependency sources are supported by uv:
+
+- [Index](#index): A package resolved from a specific package index.
+- [Git](#git): A Git repository.
+- [URL](#url): A remote wheel or source distribution.
+- [Path](#path): A local wheel, source distribution, or project directory.
+- [Workspace](#workspace-member): A member of the current workspace.
+
+!!! important
+
+    Sources are only respected by uv. If another tool is used, only the definitions in the standard
+    project tables will be used. If another tool is being used for development, any metadata
+    provided in the source table will need to be re-specified in the other tool's format.
```

Comment:
We should talk about when you need `tool.uv.sources` and that we're doing this due to holes in the spec (no relative paths, no workspace support)

Reply:
That's included above

> Dependency sources add support common patterns that are not supported by the `project.dependencies`
standard, like editable installations and relative paths. For example, to install `foo` from a
directory relative to the project root: